### PR TITLE
Adding sizing options to Badge

### PR DESCRIPTION
**Pull request by gjones**
### Summary
We may need a small size badge to fit inside text/select fields in the future, this PR does that and updates the story to allow users to toggle.

#### Example

CleanShot 2023-07-20 at 11 34 34

### CODE REVIEW

File: src/components/Badge/Badge.tsx [hunk 1]
```diff
@@ -7,25 +7,30 @@ export type BadgeState =
...
-  font: ${props => props.theme.click.badge.typography.label.default};
+  font: ${({ size = "md", theme }) => 
+    theme.click.badge.typography.label[size].default};
   border-radius: ${props => props.theme.click.badge.radii.all};
   border: ${({ state = "default", theme }) =>
     `${theme.click.badge.stroke} solid ${theme.click.badge.color.stroke[state]}`};
-  padding: ${props =>
-    `${props.theme.click.badge.space.y} ${props.theme.click.badge.space.x}`};
+  padding: ${({ size = "md", theme }) => 
+    theme.click.badge.space[size].y} ${({ size = "md", theme }) => theme.click.badge.space[size].x};
```
Comment (vineethasok) [inline src/components/Badge/Badge.tsx:30]: Need to add gap and display flex to this container

File: src/styles/types.ts [hunk 2]
```diff
@@ -612,17 +639,19 @@
           "default": string,
           "active": string,
           "disabled-checked": string,
```
Comment (vineethasok) [inline src/styles/types.ts:641]: Can we rename it as disabled instead of disabled-checked or unchecked

Comment (gjones) [reply src/styles/types.ts:641]: They're different states though, are we able to indicate the difference between `disabled + checked` and `disabled + unchecked` without them?

Comment (vineethasok) [reply src/styles/types.ts:641]: yes we are able to differentiate with the data attribute  `[data-state] | "checked" \| "unchecked" \| "indeterminate"`

Comment (vineethasok) [reply src/styles/types.ts:641]: I asked because we have removed the disabled-unchecked from tokens

**Review by vineethasok — APPROVED** [review]

**Pull request merged**